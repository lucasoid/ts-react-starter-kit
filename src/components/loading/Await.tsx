import * as React from 'react';

interface IAwaitProps {
    loaded: boolean;
    renderFallback: Function;
    children?: any;
}

export const Await: React.FC<IAwaitProps> = props => {
    return props.loaded ? props.children : props.renderFallback();
};
