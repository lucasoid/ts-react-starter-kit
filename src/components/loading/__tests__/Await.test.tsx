import * as React from 'react';
import { shallow } from 'enzyme';
import { Await } from '../Await';

describe('Await', () => {
    it('displays a fallback UI while in a loading state', () => {
        const renderFallback = () => 'My fallback';
        let wrapper = shallow(
            <Await renderFallback={renderFallback} loaded={false}>
                My content
            </Await>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('displays the complete UI when in a loaded state', () => {
        const renderFallback = () => 'My fallback';
        let wrapper = shallow(
            <Await renderFallback={renderFallback} loaded={true}>
                My content
            </Await>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
