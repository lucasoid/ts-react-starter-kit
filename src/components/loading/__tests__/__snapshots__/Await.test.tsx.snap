// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Await displays a fallback UI while in a loading state 1`] = `"My fallback"`;

exports[`Await displays the complete UI when in a loaded state 1`] = `"My content"`;
